tests/amr/data/particles/schedule/test_tag_strategy.h

```diff
@@ -79,29 +78,44 @@ class TagStrategy : public SAMRAI::mesh::StandardTagAndInitStrategy
 
                     auto particlesBox = particlesData->getBox();
 
-                    // here we are 1D
-                    for (int iCellPos = particlesBox.lower(dirX);
-                         iCellPos <= particlesBox.upper(dirX); ++iCellPos)
+                    auto lowerZ = dimension == 3 ? particlesBox.lower(dirZ) : 0;
+                    auto upperZ = dimension == 3 ? particlesBox.upper(dirZ) : 0;
+
+                    auto lowerY = dimension == 2 ? particlesBox.lower(dirY) : 0;
+                    auto upperY = dimension == 2 ? particlesBox.upper(dirY) : 0;
+
+                    auto lowerX = particlesBox.lower(dirX);
+                    auto upperX = particlesBox.upper(dirX);
+
+                    for (auto iCellZ = lowerZ; iCellZ <= upperZ; ++iCellZ)
                     {
-                        float middle = 0.5;
-                        float delta  = 0.30;
+                        for (auto iCellY = lowerY; iCellY <= upperY; ++iCellY)
+                        {
+                            for (auto iCellX = lowerX; iCellX <= upperX; ++iCellX)
+                            {
+                                std::array<int, 3> _3DiCell = {iCellX, iCellY, iCellZ};
+                                float middle                = 0.5;
+                                float delta                 = 0.30;
 
-                        Particle<dimension> particle;
+                                Particle<dimension> particle;
 
-                        particle.weight = 1.;
-                        particle.charge = 1.;
-                        particle.v      = {{1.0, 0.0, 0.0}};
+                                for (size_t i = 0; i < dimension; i++)
+                                    particle.iCell[i] = _3DiCell[i];
 
-                        particle.iCell[dirX] = iCellPos;
+                                particle.delta.fill(middle);
```

Comment:
Im' not sure it matters for these tests where the particles are exactly, I didn't look... but I don't think we want to put `dimension` times a particle in the middle of the cell, that's weird. Probably better to dispatch them in distinct places.

Reply:
> I don't think we want to put dimension times a particle in the middle of the cell

I don't understand this.

> Probably better to dispatch them in distinct places.

I would state that they are.

Reply:
I misread sorry